game-service/src/main/java/org/libertybikes/game/core/GameBoard.java

```diff
@@ -139,4 +142,58 @@ public boolean moveObjects() {
         return true;
     }
 
+    public boolean broadcastToAI() {
+        if (ai.isEmpty()) {
+            return false;
+        }
+
+        for (AIPlayer p : ai) {
+            try {
+                p.broadCastBoard(board);
```

Comment:
We can't pass the actual board to AI, because that would allow them to set spaces on the board (e.g. surround opponents with walls, or clear a path for themselves by hand).  Instead, we need to pass in a read-only version of the array to the AI.